docs/data/tree-view/simple-tree-view/customization/GmailTreeView.js

```diff
@@ -13,85 +13,129 @@ import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
-    },
-    '&:hover': {
-      backgroundColor: theme.palette.action.hover,
-    },
-    '&.Mui-focused, &.Mui-selected, &.Mui-selected.Mui-focused': {
-      backgroundColor: `var(--tree-view-bg-color, ${theme.palette.action.selected})`,
-      color: 'var(--tree-view-color)',
-    },
-    [`& .${treeItemClasses.label}`]: {
-      fontWeight: 'inherit',
-      color: 'inherit',
-    },
-    [`& .${treeItemClasses.iconContainer}`]: {
-      marginRight: theme.spacing(1),
-    },
+}));
+
+const CustomTreeItemContent = styled(TreeItemNextContent)(({ theme }) => ({
+  marginBottom: theme.spacing(0.3),
+  color: theme.palette.text.secondary,
+  borderRadius: theme.spacing(2),
+  paddingRight: theme.spacing(1),
+  fontWeight: theme.typography.fontWeightMedium,
+  '&.expanded': {
+    fontWeight: theme.typography.fontWeightRegular,
+  },
+  '&:hover': {
+    backgroundColor: theme.palette.action.hover,
   },
-  [`& .${treeItemClasses.group}`]: {
+  '&.focused, &.selected, &.selected.focused': {
+    backgroundColor: `var(--tree-view-bg-color, ${theme.palette.action.selected})`,
+    color: 'var(--tree-view-color)',
+  },
+}));
+
+const CustomTreeItemIconContainer = styled(TreeItemNextIconContainer)(
+  ({ theme }) => ({
+    marginRight: theme.spacing(1),
+  }),
+);
+
+const CustomTreeItemTransitionGroup = styled(TreeItemNextTransitionGroup)(
+  ({ theme }) => ({
     marginLeft: 0,
-    [`& .${treeItemClasses.content}`]: {
+    [`& .content`]: {
       paddingLeft: theme.spacing(2),
     },
-  },
-}));
+  }),
+);
 
-const StyledTreeItem = React.forwardRef(function StyledTreeItem(props, ref) {
+const CustomTreeItem = React.forwardRef(function CustomTreeItem(props, ref) {
   const theme = useTheme();
   const {
+    id,
+    nodeId,
+    label,
+    children,
+    slots,
+    slotProps,
     bgColor,
     color,
     labelIcon: LabelIcon,
     labelInfo,
-    labelText,
     colorForDarkMode,
     bgColorForDarkMode,
     ...other
   } = props;
 
-  const styleProps = {
+  const {
+    getRootProps,
+    getContentProps,
+    getIconContainerProps,
+    getLabelProps,
+    getGroupTransitionProps,
+    status,
+  } = useTreeItem({ id, nodeId, children, label, rootRef: ref });
+
+  const style = {
     '--tree-view-color': theme.palette.mode !== 'dark' ? color : colorForDarkMode,
     '--tree-view-bg-color':
       theme.palette.mode !== 'dark' ? bgColor : bgColorForDarkMode,
   };
 
   return (
-    <StyledTreeItemRoot
-      label={
-        <Box
-          sx={{
-            display: 'flex',
-            alignItems: 'center',
-            p: 0.5,
-            pr: 0,
-          }}
+    <TreeItemProvider nodeId={nodeId}>
+      <CustomTreeItemRoot {...getRootProps({ ...other, style })}>
+        <CustomTreeItemContent
+          {...getContentProps({
+            className: clsx('content', {
+              expanded: status.expanded,
+              selected: status.selected,
+              focused: status.focused,
+            }),
```

Comment:
Out of context: This kind of behavior will need some educational efforts.

I went to [base docs](https://mui.com/base-ui/getting-started/customization/#creating-custom-components-using-hooks) to check what `getXxxxProps` take as an argument. I assume other dev not familiar with with base might wonder what are those functions

Reply:
Yep, I don't want to rush the merging of this PR because it clearly needs to come with some good doc on the Base UI side and backlinks to it on our side.

Reply:
FYI, as we're [changing the API](https://www.notion.so/mui-org/base-ui-Customization-API-3996f854a02943ca8c34ccbaac80e6dc) of the Base UI components and hooks, we're limiting the contributions to the current version of the library. Going forward, Base UI will be developed in its own repo (but we won't remove the existing sources from the Core repo, as it's used in several places).